### DISCUSSION TITLE
Runner registration token API - PAT Scope

### DISCUSSION THREAD

**Discussion by RobinFrcd** [community/community#30805] [Actions]
Hi,
I'm trying to fully automatize the deployment of new tokens in my cluster. To do so, we currently need to create a [registration token](https://docs.github.com/en/rest/actions/self-hosted-runners#create-a-registration-token-for-an-organization) through the API.

The thing is, to generate this token we need a PAT with `admin:org` scope which seems far too broad to just register new runners. I really want to avoid having one of these secrets being in the wild, even if it's a protected secret. 

Why isn't it possible to give a more restrictive scope to the PAT only to register a runner ?

Thanks

**Comment by dgholz** [comment]
PATs aren't great for restricting scope: they're supposed to be in-place of the user clicking in the GitHub UI, so an 'admin' user should see all the Admin actions.

I recommend using a GitHub App instead, and storing the secrets securely. Use them [to generate a JWT to then request an installation token](https://docs.github.com/en/developers/apps/building-github-apps/authenticating-with-github-apps#authenticating-as-an-installation) that can be used to create a registration token. A lot more work! But much more secure (the JWT + installation tokens expire very quickly), and GitHub App permissions are very fine-grained: you can create an app just for registering runners, and install it onto just the repositories that need them.

This is exactly what we're doing, we have a GitHub App without running a service & just use its secrets to register runners.
